File: app/src/main/java/org/test/kotlin/demo/repos/ReposModule.java
package org.test.kotlin.demo.repos;

import javax.inject.Named;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public interface ReposModule {

    @Binds
    ReposContract.View bindView(ReposFragment impl);

    @Binds
    ReposContract.Presenter bindPresenter(ReposPresenter impl);

    @Provides
    @Named(ReposContract.ARG_USERNAME)
    static String provideUserName(ReposFragment fragment) {
        return fragment.getArguments().getString(ReposContract.ARG_USERNAME);
    }

}

File: app/src/main/java/org/test/kotlin/demo/repos/ReposModule.kt
package org.test.kotlin.demo.repos

import dagger.Binds
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
abstract class ReposModule {

    @Binds
    abstract fun bindView(impl: ReposFragment): ReposContract.View

    @Binds
    abstract fun bindPresenter(impl: ReposPresenter): ReposContract.Presenter

    @Module
    companion object {

        @Provides
        @JvmStatic
        @Named(ReposContract.ARG_USERNAME)
        fun provideUserName(fragment: ReposFragment) =
                fragment.arguments!!.getString(ReposContract.ARG_USERNAME)!!

    }

}
